### PR TITLE
fix: ensure chrome data dir is none when appropiate

### PR DESCRIPTION
**Pull request by apkallum**
# Summary

fix: ensure chrome data dir is none when appropiate

# Changes these areas

- [x] Bugfixes
- [ ] Feature behavior
- [ ] Command line interface
- [ ] Configuration options
- [ ] Internal architecture
- [ ] Archived data layout on disk

# Roadmap Goals

### CODE REVIEW

File: archivebox/config/__init__.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def get_real_name(key: str) -> str:
     'LOGS_DIR':                 {'default': lambda c: c['OUTPUT_DIR'] / LOGS_DIR_NAME},
     'CONFIG_FILE':              {'default': lambda c: Path(c['CONFIG_FILE']).resolve() if c['CONFIG_FILE'] else c['OUTPUT_DIR'] / CONFIG_FILENAME},
     'COOKIES_FILE':             {'default': lambda c: c['COOKIES_FILE'] and Path(c['COOKIES_FILE']).resolve()},
-    'CHROME_USER_DATA_DIR':     {'default': lambda c: find_chrome_data_dir() if c['CHROME_USER_DATA_DIR'] is None else Path(c['CHROME_USER_DATA_DIR']).resolve() or None},
+    'CHROME_USER_DATA_DIR':     {'default': lambda c: find_chrome_data_dir() if c['CHROME_USER_DATA_DIR'] is None else Path(c['CHROME_USER_DATA_DIR']).resolve() if Path(c['CHROME_USER_DATA_DIR']).resolve() != Path.cwd() else None},
```
Comment (cdvv7788) [inline archivebox/config/__init__.py:235]: This should work. Another option would be checking against the value without touching paths: `if c['CHROME_USER_DATA_DIR'] else None`.
@pirate which do you prefer?

Comment (pirate) [reply archivebox/config/__init__.py:235]: I prefer `if c['CHROME_USER_DATA_DIR'] else None`

File: archivebox/config/__init__.py [hunk 2]
```diff
@@ -232,7 +232,7 @@ def get_real_name(key: str) -> str:
     'LOGS_DIR':                 {'default': lambda c: c['OUTPUT_DIR'] / LOGS_DIR_NAME},
     'CONFIG_FILE':              {'default': lambda c: Path(c['CONFIG_FILE']).resolve() if c['CONFIG_FILE'] else c['OUTPUT_DIR'] / CONFIG_FILENAME},
     'COOKIES_FILE':             {'default': lambda c: c['COOKIES_FILE'] and Path(c['COOKIES_FILE']).resolve()},
-    'CHROME_USER_DATA_DIR':     {'default': lambda c: find_chrome_data_dir() if c['CHROME_USER_DATA_DIR'] is None else Path(c['CHROME_USER_DATA_DIR']).resolve() or None},
+    'CHROME_USER_DATA_DIR':     {'default': lambda c: find_chrome_data_dir() if c['CHROME_USER_DATA_DIR'] is None else Path(c['CHROME_USER_DATA_DIR']).resolve() if c['CHROME_USER_DATA_DIR']) else None},
```
Comment (pirate) [inline archivebox/config/__init__.py:235]:
```suggestion
    'CHROME_USER_DATA_DIR':     {'default': lambda c: find_chrome_data_dir() if c['CHROME_USER_DATA_DIR'] is None else Path(c['CHROME_USER_DATA_DIR']).resolve() if c['CHROME_USER_DATA_DIR']) else None},   # None means unset, so we autodetect it with find_chrome_Data_dir(), but emptystring '' means user manually set it to '', and we should store it as None
```

File: archivebox/config/__init__.py [hunk 3]
```diff
@@ -232,7 +232,7 @@ def get_real_name(key: str) -> str:
     'LOGS_DIR':                 {'default': lambda c: c['OUTPUT_DIR'] / LOGS_DIR_NAME},
     'CONFIG_FILE':              {'default': lambda c: Path(c['CONFIG_FILE']).resolve() if c['CONFIG_FILE'] else c['OUTPUT_DIR'] / CONFIG_FILENAME},
     'COOKIES_FILE':             {'default': lambda c: c['COOKIES_FILE'] and Path(c['COOKIES_FILE']).resolve()},
-    'CHROME_USER_DATA_DIR':     {'default': lambda c: find_chrome_data_dir() if c['CHROME_USER_DATA_DIR'] is None else Path(c['CHROME_USER_DATA_DIR']).resolve() if c['CHROME_USER_DATA_DIR']) else None},
+    'CHROME_USER_DATA_DIR':     {'default': lambda c: find_chrome_data_dir() if c['CHROME_USER_DATA_DIR'] is None else Path(c['CHROME_USER_DATA_DIR']).resolve() if c['CHROME_USER_DATA_DIR']) else None},   # None means unset, so we autodetect it with find_chrome_Data_dir(), but emptystring '' means user manually set it to '', and we should store it as None
```
Comment (ttimasdf) [inline archivebox/config/__init__.py:235]: Theres a typo here `if c['CHROME_USER_DATA_DIR']) else None`.

one more `)`

**Pull request merged**